kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaCommandDecoder.java

```diff
@@ -67,7 +64,7 @@
     public KafkaCommandDecoder(StatsLogger statsLogger, KafkaServiceConfiguration kafkaConfig) {
         this.requestStats = new RequestStats(statsLogger);
         this.kafkaConfig = kafkaConfig;
-        this.requestQueue = new FakeArrayBlockingQueue(kafkaConfig.getMaxQueuedRequests());
+        this.requestQueue = new LinkedBlockingQueue<>(kafkaConfig.getMaxQueuedRequests());
```

Comment:
ArrayBlockingQueue maybe more effective

Reply:
Kafka also uses `LinkedBlockingDeque` for `responseQueue`, see

https://github.com/apache/kafka/blob/ad91c5edf09af83cf44e2be71f19407ade1896bb/core/src/main/scala/kafka/network/SocketServer.scala#L823

(BTW, I don't understand why it uses deque but not queue. Only `put` and `poll` methods are used)

Here I concern more about the memory usage over performance. `ArrayBlockingQueue` allocates 500 (the default `maxQueuedRequests` value) objects when it's constructed. For N channels, there're 500*N objects that were allocated on heap. Maybe it looks like not very much, but it's still a waste.